pkg/controller/restore_controller.go

```diff
@@ -464,32 +475,8 @@ func mostRecentCompletedBackup(backups []*api.Backup) *api.Backup {
 
 func (c *restoreController) fetchBackup(objectStore cloudprovider.ObjectStore, name string) (*api.Backup, error) {
```

Comment:
if we decide to remove the "fallback to obj storage" functionality, then we can get rid of this function entirely

Reply:
I'd recommend reviewing https://github.com/heptio/ark/issues/42

Reply:
That's a helpful refresh.  My preference would be to make the sync process more intelligent -- off the top of my head, we could run it much more frequently, but have some kind of checksum or index file within the backup locations so the sync can easily detect whether or not things have changed

Reply:
I read the referred issue with a fresh mind today. At the cost of speaking the obvious, seems if we just remove the fallback then we'd be regressing and running into that undesirable use case.

I'd think doing a diff on a checksum even if very frequently should be comfortably low cost. My question at the moment is, would adding the operation to do a checksum in a backup add any significant processing cost to a single operation, and would the operations happen too frequently to make this an issue? Or we need a different (daemon) process to run thru the backups and continuously spit out the checksum somewhere? Things that I'm thinking about now.

Reply:
Mulled this over the weekend, and I'm thinking we should just update this code with the naive implementation for now (if backup isn't found via Kube API, then check *each* backup location, starting with the default), and look to do an improvement on the backup sync controller as part of the "backup sharing" epic